agent/internal/task_config.go

```diff
@@ -133,41 +133,81 @@ func (c *TaskConfig) GetCloneMethod() string {
 }
 
 func (tc *TaskConfig) GetTaskGroup(taskGroup string) (*model.TaskGroup, error) {
-	if tc == nil {
-		return nil, errors.New("unable to get task group because task config is nil")
-	}
-	if tc.Task == nil {
-		return nil, errors.New("unable to get task group because task is nil")
-	}
-	if tc.Task.Version == "" {
-		return nil, errors.New("task has no version")
-	}
-	if tc.Project == nil {
-		return nil, errors.New("project is nil")
+	if err := tc.validateTaskConfig(); err != nil {
+		return nil, err
 	}
 
-	var tg *model.TaskGroup
 	if taskGroup == "" {
-		// if there is no named task group, fall back to project definitions
-		tg = &model.TaskGroup{
-			SetupTask:               tc.Project.Pre,
-			TeardownTask:            tc.Project.Post,
-			Timeout:                 tc.Project.Timeout,
-			SetupGroupFailTask:      tc.Project.Pre == nil || tc.Project.PreErrorFailsTask,
-			TeardownTaskCanFailTask: tc.Project.Post == nil || tc.Project.PostErrorFailsTask,
-		}
-	} else {
-		tg = tc.Project.FindTaskGroup(taskGroup)
-		if tg == nil {
-			return nil, errors.Errorf("couldn't find task group '%s' in project '%s'", tc.Task.TaskGroup, tc.Project.Identifier)
-		}
+		return nil, nil
 	}
+	tg := tc.Project.FindTaskGroup(taskGroup)
+	if tg == nil {
+		return nil, errors.Errorf("couldn't find task group '%s' in project '%s'", tc.Task.TaskGroup, tc.Project.Identifier)
```

Comment:
nit: It seems like we should use `taskGroup` in the error format rather than reference `tc.Task.TaskGroup` to be consistent.